Function to create custom image locally and test to validate

keesturam opened the pull request:
Fixes #403

keesturam commented on tests/helpers.py:
```diff
@@ -854,6 +858,99 @@ def create_pods(pvc_objs, pod_factory, interface, pods_for_rwx=1, status=""):
+        oc_cmd += f"--kubeconfig {kubeconfig} "
+    oc_cmd += command
+    logger.info(f'Running command {oc_cmd}')
+    result = run(
```
Usage of subprocess.run here is intentional as the existing function `run_cmd` uses `shlex.split` before sending the command, which removes `"` in the command being sent.

RazTamir replied:
there are other ways to fix this bug +1 on this usage

ramkiperiy commented on tests/helpers.py:
```diff
@@ -854,6 +858,99 @@ def create_pods(pvc_objs, pod_factory, interface, pods_for_rwx=1, status=""):
+        install_package (str): package to install over the base image
+
+    Returns:
+        (obj): Returns the image object
```
Please add the exceptions, also will be good if you can add more comments in the code.

Avilir commented on tests/helpers.py:
```diff
@@ -854,6 +858,99 @@ def create_pods(pvc_objs, pod_factory, interface, pods_for_rwx=1, status=""):
+        install_package,
+        namespace,
+        source_image='centos',
+        source_image_label='latest'
```
why dont use ubi8 as default image ?
it is much smaller

keesturam replied:
The functionality is to make a build by passing a docker file. I believe it looks for the base image from the local image stream. I don't see a standard ubi8 image available.

`error: unable to locate any images in image streams, local docker images with name "ubi:8"`

But this is a valid point, I'll try to see if there is a leaner image available. I'll make that change as part of a different PR.

Avilir replied:
currect, i am using it but it located at : registry.access.redhat.com/ubi8

Avilir commented on tests/helpers.py:
```diff
@@ -854,6 +858,99 @@ def create_pods(pvc_objs, pod_factory, interface, pods_for_rwx=1, status=""):
+        stdout=PIPE,
+        stderr=PIPE,
+        timeout=15,
+        shell=True
```
Not sure that 15 sec. is enough time to build the image. you may need to increase this time .

keesturam replied:
This command doesn't wait for the image to be built. A build job is just started.
The command just passes or fails almost immediately.

RazTamir commented on tests/e2e/test_pod_from_registry.py:
```diff
@@ -0,0 +1,41 @@
+            image_name='fio',
+            install_package='fio'
+        )
+        image_id = image_obj.get()['status']['tags'][0]['items'][0]['dockerImageReference']
```
please use .get() for all

keesturam replied:
Done.

RazTamir commented on tests/e2e/test_pod_from_registry.py:
```diff
@@ -0,0 +1,41 @@
+        )
+        image_id = image_obj.get()['status']['tags'][0]['items'][0]['dockerImageReference']
+        pod_dict = templating.load_yaml(constants.CSI_CEPHFS_POD_YAML)
+        pod_dict['spec']['containers'][0]['image'] = image_obj.resource_name
```
same

keesturam replied:
done

RazTamir commented on tests/helpers.py:
```diff
@@ -854,6 +858,99 @@ def create_pods(pvc_objs, pod_factory, interface, pods_for_rwx=1, status=""):
+        install_package (str): package to install over the base image
+
+    Returns:
+        (obj): Returns the image object
```
OCP object?

keesturam replied:
Yes, updated now.

ebenahar commented on tests/helpers.py:
```diff
@@ -873,6 +877,101 @@ def create_pods(pvc_objs, pod_factory, interface, pods_for_rwx=1, status=""):
+
+    """
+    base_image = source_image + ':' + source_image_label
+    docker_file = f"FROM {base_image}\n" \
```
please use parentheses in order to break lines

keesturam replied:
Done

keesturam commented on tests/e2e/test_pod_from_registry.py:
```diff
@@ -0,0 +1,41 @@
+logger = logging.getLogger(__name__)
+
+
+@ignore_leftovers
```
This is not required. Removed it.

keesturam commented:
Test run locally after addressing review comments. 
```
run-ci tests/e2e/workloads/test_pod_from_registry.py --cluster-path /home/kramdoss/Downloads/kramdoss-cluster --cluster-name kramdoss-vmware-cluster
Collecting Cluster versions
Problem ocurred when looking for some resource! Error: Requested packageManifest: ocs-operator with selector: ocs-operator-internal=true not found!
Failed to gather version details! The report of version mightnot be complete!
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.7.6, pytest-5.3.2, py-1.8.0, pluggy-0.13.1
rootdir: /home/kramdoss/git/ocs-ci, inifile: pytest.ini
plugins: reportportal-1.0.5, logger-0.5.1, metadata-1.8.0, html-2.0.1, marker-bugzilla-0.9.3, ordering-0.6
collected 1 item                                                                                                                                                                                                  

tests/e2e/workloads/test_pod_from_registry.py::TestRegistryImage::test_run_pod_local_image 
------------------------------------------------------------------------------------------------- live log setup --------------------------------------------------------------------------------------------------
15:01:16 - MainThread - tests.conftest - INFO - All logs located at /tmp/ocs-ci-logs-1582709466
15:01:16 - MainThread - ocs_ci.deployment.factory - INFO - Deployment key = aws_ipi
15:01:16 - MainThread - ocs_ci.deployment.factory - INFO - Current deployment platform: AWS,deployment type: ipi
15:01:18 - MainThread - ocs_ci.utility.utils - INFO - Executing command: /home/kramdoss/git/ocs-ci/bin/oc version --client
15:01:18 - MainThread - ocs_ci.utility.utils - INFO - OpenShift Client version: Client Version: openshift-clients-4.2.2-201910250432-12-g72076900

15:01:18 - MainThread - ocs_ci.ocs.version - INFO - collecting ocs version
15:01:18 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get clusterversion version -o yaml
15:01:19 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get namespace  -o yaml
15:01:20 - MainThread - ocs_ci.ocs.version - INFO - found storage namespaces ['openshift-cluster-storage-operator', 'openshift-storage']
15:01:20 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-cluster-storage-operator --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get pod  -n openshift-cluster-storage-operator -o yaml
15:01:20 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get pod  -n openshift-storage -o yaml
15:01:22 - MainThread - ocs_ci.ocs.version - INFO - ClusterVersion .spec.channel: stable-4.3
15:01:22 - MainThread - ocs_ci.ocs.version - INFO - ClusterVersion .status.desired.version: 4.3.0-0.nightly-2020-02-12-232004
15:01:22 - MainThread - ocs_ci.ocs.version - INFO - ClusterVersion .status.desired.image: registry.svc.ci.openshift.org/ocp/release@sha256:176e814966be8903410ff23e3938216dcb356ea94e643355514a7e9ff9189825
15:01:22 - MainThread - ocs_ci.ocs.version - INFO - storage namespace openshift-cluster-storage-operator
15:01:22 - MainThread - ocs_ci.ocs.version - INFO - image quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:88da9e19713bf11e07043d3a0414f004386627aa69d08fa31e6fe4d95d67d94d {'quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:88da9e19713bf11e07043d3a0414f004386627aa69d08fa31e6fe4d95d67d94d'}
15:01:22 - MainThread - ocs_ci.ocs.version - INFO - storage namespace openshift-storage
15:01:22 - MainThread - ocs_ci.ocs.version - INFO - image quay.io/screeley44/aws-s3-provisioner:v1.0.0 {'quay.io/screeley44/aws-s3-provisioner@sha256:7ea6b604280f036b0d2f75a3337c8f1fae37cbf2fe85dc5cd423a858ecb5df31'}
15:01:22 - MainThread - ocs_ci.ocs.version - INFO - image quay.io/rhceph-dev/cephcsi:4.3-10.163d208f.release_4.3 {'quay.io/rhceph-dev/cephcsi@sha256:4120168ac4a73e33d74ebfc411374e4b965127c2af8aa220de304770479471d5'}
15:01:22 - MainThread - ocs_ci.ocs.version - INFO - image registry.redhat.io/openshift4/ose-csi-driver-registrar:4.3 {'registry.redhat.io/openshift4/ose-csi-driver-registrar@sha256:4f1b72fcefea8fc44e3fcfc5db55540e88711d84fc27b729c3e6df16f16dc9fd'}
15:01:22 - MainThread - ocs_ci.ocs.version - INFO - image registry.redhat.io/openshift4/ose-csi-external-attacher:4.3 {'registry.redhat.io/openshift4/ose-csi-external-attacher@sha256:eac36b9f204dc65b60092235d7edc9b7082489d78a4b6a485e0a17e6f01fc5cf'}
15:01:22 - MainThread - ocs_ci.ocs.version - INFO - image registry.redhat.io/openshift4/ose-csi-external-provisioner-rhel7:4.3 {'registry.redhat.io/openshift4/ose-csi-external-provisioner-rhel7@sha256:c0bfe783b63ace7244d8e06adcdad3dccd8b16eb04f56b398302c7910d2b58b3'}
15:01:22 - MainThread - ocs_ci.ocs.version - INFO - image quay.io/k8scsi/csi-resizer:v0.4.0 {'quay.io/k8scsi/csi-resizer@sha256:6883482c4c4bd0eabb83315b5a9e8d9c0f34357980489a679a56441ec74c24c9'}
15:01:22 - MainThread - ocs_ci.ocs.version - INFO - image quay.io/rhceph-dev/mcg-core:5.3.0-9.b2464eca4.5.3 {'quay.io/rhceph-dev/mcg-core@sha256:12f0e8a7c4dd8315a1a52589adb84e353e8fe39a171751e8ba5922476aa26e20'}
15:01:22 - MainThread - ocs_ci.ocs.version - INFO - image registry.redhat.io/rhscl/mongodb-36-rhel7:latest {'registry.redhat.io/rhscl/mongodb-36-rhel7@sha256:856c0c3e3b7a71795262e29915bf748ee692ecc6dcef0ef3eff0dd1fdb2f85f7'}
15:01:22 - MainThread - ocs_ci.ocs.version - INFO - image quay.io/rhceph-dev/mcg-operator:2.1.0-10.a7d9544.2.1 {'quay.io/rhceph-dev/mcg-operator@sha256:5408d535315e4b492d4904c645c94a5dce71e29632ffb48e1c12e4c6695a2681'}
15:01:22 - MainThread - ocs_ci.ocs.version - INFO - image quay.io/rhceph-dev/ocs-operator:4.3-8.8ec8be3.release_4.3 {'quay.io/rhceph-dev/ocs-operator@sha256:f4b80ba681bd1b342c703a96ed584a7c27a2893845188ef59e7db44a9650c8ef'}
15:01:22 - MainThread - ocs_ci.ocs.version - INFO - image quay.io/rhceph-dev/rook-ceph:4.3-15.7c6c68c8.ocs_4.3 {'quay.io/rhceph-dev/rook-ceph@sha256:9a546adc47d0397451ad09561dcc70db41e24e10fcbdc23890e780adebf19b0d'}
15:01:22 - MainThread - ocs_ci.ocs.version - INFO - image quay.io/rhceph-dev/rhceph:4-15 {'quay.io/rhceph-dev/rhceph@sha256:0279e891701e0f97efbae1dd5c3f7255ae857bd093be80d3c02006425c9a8e5a'}
15:01:22 - MainThread - tests.conftest - INFO - human readable ocs version info written into /home/kramdoss/Downloads/kramdoss-cluster/ocs_version.2020-02-26T15:01:22.691956
15:01:22 - MainThread - ocs_ci.utility.utils - INFO - testrun_name: OCS4-3-Downstream-OCP4-3-AWS-IPI-3AZ-RHCOS-3M-3W-
15:01:22 - MainThread - ocs_ci.utility.utils - INFO - The rook directory /home/kramdoss/git/ocs-ci/data/rook already exists, ocs-ci will skip the clone of rook repository.
15:01:22 - MainThread - ocs_ci.utility.utils - INFO - Fetching latest changes from rook repository.
15:01:22 - MainThread - ocs_ci.utility.utils - INFO - Executing command: git fetch --all
15:01:24 - MainThread - ocs_ci.utility.utils - INFO - Checkout rook repository to specific branch: master
15:01:24 - MainThread - ocs_ci.utility.utils - INFO - Executing command: git checkout master
15:01:24 - MainThread - ocs_ci.utility.utils - WARNING - Command warning: Already on 'master'

15:01:24 - MainThread - ocs_ci.utility.utils - INFO - Reset branch: master with latest changes
15:01:24 - MainThread - ocs_ci.utility.utils - INFO - Executing command: git reset --hard origin/master
15:01:24 - Dummy-1 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod  -A -o yaml
15:01:24 - Dummy-2 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get StorageClass  -A -o yaml
15:01:24 - Dummy-3 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get CephFileSystem  -A -o yaml
15:01:24 - Dummy-4 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get CephBlockPool  -A -o yaml
15:01:24 - Dummy-5 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get PersistentVolume  -A -o yaml
15:01:24 - Dummy-6 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get PersistentVolumeClaim  -A -o yaml
15:01:24 - Dummy-7 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Namespace  -A -o yaml
-------------------------------------------------------------------------------------------------- live log call --------------------------------------------------------------------------------------------------
15:01:31 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc new-project namespace-test-b4970d6205b54e26a90a202cf2ab4895
15:01:32 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get storageclass ocs-storagecluster-ceph-rbd -o yaml
15:01:33 - MainThread - ocs_ci.ocs.resources.ocs - INFO - Adding PersistentVolumeClaim with name pvc-test-83e89e48b28444198e0597f199725708
15:01:33 - MainThread - ocs_ci.utility.templating - INFO - apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: pvc-test-83e89e48b28444198e0597f199725708
  namespace: namespace-test-b4970d6205b54e26a90a202cf2ab4895
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 5Gi
  storageClassName: ocs-storagecluster-ceph-rbd

15:01:33 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig create -f /tmp/PersistentVolumeClaimom0yewb1 -o yaml
15:01:34 - MainThread - ocs_ci.ocs.ocp - INFO - Waiting for a resource(s) of kind PersistentVolumeClaim identified by name 'pvc-test-83e89e48b28444198e0597f199725708' using selector None at column name STATUS to reach desired condition Bound
15:01:34 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get PersistentVolumeClaim pvc-test-83e89e48b28444198e0597f199725708 -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 -o yaml
15:01:35 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get PersistentVolumeClaim pvc-test-83e89e48b28444198e0597f199725708 -n namespace-test-b4970d6205b54e26a90a202cf2ab4895
15:01:35 - MainThread - tests.helpers - INFO - PersistentVolumeClaim pvc-test-83e89e48b28444198e0597f199725708 reached state Bound
15:01:35 - MainThread - tests.helpers - INFO - Running command oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig new-build -D $'FROM centos:7
 RUN yum install -y fio
 CMD tail -f /dev/null' --name=fio
15:01:44 - MainThread - tests.helpers - INFO - --> Found container image 5e35e35 (3 months old) from Docker Hub for "centos:7"

    * An image stream tag will be created as "centos:7" that will track the source image
    * A Docker build using a predefined Dockerfile will be created
      * The resulting image will be pushed to image stream tag "fio:latest"
      * Every time "centos:7" changes a new build will be triggered

--> Creating resources with label build=fio ...
    imagestream.image.openshift.io "centos" created
    imagestream.image.openshift.io "fio" created
    buildconfig.build.openshift.io "fio" created
--> Success

15:01:44 - MainThread - ocs_ci.ocs.ocp - INFO - Waiting for a resource(s) of kind Pod identified by name 'fio-1-build' using selector None at column name STATUS to reach desired condition Completed
15:01:44 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod fio-1-build -o yaml
15:01:44 - MainThread - ocs_ci.ocs.ocp - WARNING - Failed to get resource: fio-1-build of kind: Pod, selector: None, Error: Error during execution of command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod fio-1-build -o yaml.
Error is Error from server (NotFound): pods "fio-1-build" not found

15:01:44 - MainThread - ocs_ci.ocs.ocp - WARNING - Number of attempts to get resource reached!
15:01:44 - MainThread - root - ERROR - Exception raised during iteration: Error during execution of command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod fio-1-build -o yaml.
Error is Error from server (NotFound): pods "fio-1-build" not found

15:01:44 - MainThread - ocs_ci.utility.utils - INFO - Going to sleep for 30 seconds before next iteration
15:02:14 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod fio-1-build -o yaml
15:02:16 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod fio-1-build
15:02:16 - MainThread - ocs_ci.ocs.ocp - INFO - status of fio-1-build at column STATUS was Running, but we were waiting for Completed
15:02:16 - MainThread - ocs_ci.utility.utils - INFO - Going to sleep for 30 seconds before next iteration
15:02:46 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod fio-1-build -o yaml
15:02:47 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod fio-1-build
15:02:48 - MainThread - ocs_ci.ocs.ocp - INFO - status of fio-1-build at column STATUS was Running, but we were waiting for Completed
15:02:48 - MainThread - ocs_ci.utility.utils - INFO - Going to sleep for 30 seconds before next iteration
15:03:18 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod fio-1-build -o yaml
15:03:19 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod fio-1-build
15:03:20 - MainThread - ocs_ci.ocs.ocp - INFO - status of fio-1-build at column STATUS was Running, but we were waiting for Completed
15:03:20 - MainThread - ocs_ci.utility.utils - INFO - Going to sleep for 30 seconds before next iteration
15:03:50 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod fio-1-build -o yaml
15:03:51 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod fio-1-build
15:03:52 - MainThread - tests.helpers - INFO - build fio ready
15:03:52 - MainThread - tests.helpers - INFO - image lookup for image"fio" is set
15:03:52 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig set image-lookup fio
15:03:53 - MainThread - tests.helpers - INFO - image fio can now be consumed
15:03:53 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get ImageStream fio -o yaml
15:03:54 - MainThread - ocs_ci.ocs.resources.ocs - INFO - Adding Pod with name csicephfs-demo-pod
15:03:54 - MainThread - ocs_ci.utility.templating - INFO - apiVersion: v1
kind: Pod
metadata:
  name: csicephfs-demo-pod
spec:
  containers:
  - image: fio
    name: web-server
    volumeMounts:
    - mountPath: /var/lib/www/html
      name: mypvc
  volumes:
  - name: mypvc
    persistentVolumeClaim:
      claimName: pvc-test-83e89e48b28444198e0597f199725708
      readOnly: false

15:03:54 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig create -f /tmp/Podvksvcyap -o yaml
15:03:54 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod csicephfs-demo-pod -o yaml
15:03:55 - MainThread - ocs_ci.ocs.ocp - INFO - Waiting for a resource(s) of kind Pod identified by name 'csicephfs-demo-pod' using selector None at column name STATUS to reach desired condition Running
15:03:55 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod csicephfs-demo-pod -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 -o yaml
15:03:56 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod csicephfs-demo-pod -n namespace-test-b4970d6205b54e26a90a202cf2ab4895
15:03:57 - MainThread - ocs_ci.ocs.ocp - INFO - status of csicephfs-demo-pod at column STATUS was ContainerCreating, but we were waiting for Running
15:03:57 - MainThread - ocs_ci.utility.utils - INFO - Going to sleep for 3 seconds before next iteration
15:04:00 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod csicephfs-demo-pod -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 -o yaml
15:04:01 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod csicephfs-demo-pod -n namespace-test-b4970d6205b54e26a90a202cf2ab4895
15:04:01 - MainThread - ocs_ci.ocs.ocp - INFO - status of csicephfs-demo-pod at column STATUS was ContainerCreating, but we were waiting for Running
15:04:01 - MainThread - ocs_ci.utility.utils - INFO - Going to sleep for 3 seconds before next iteration
15:04:04 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod csicephfs-demo-pod -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 -o yaml
15:04:05 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod csicephfs-demo-pod -n namespace-test-b4970d6205b54e26a90a202cf2ab4895
15:04:06 - MainThread - ocs_ci.ocs.ocp - INFO - status of csicephfs-demo-pod at column STATUS was ContainerCreating, but we were waiting for Running
15:04:06 - MainThread - ocs_ci.utility.utils - INFO - Going to sleep for 3 seconds before next iteration
15:04:09 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod csicephfs-demo-pod -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 -o yaml
15:04:09 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod csicephfs-demo-pod -n namespace-test-b4970d6205b54e26a90a202cf2ab4895
15:04:10 - MainThread - ocs_ci.ocs.ocp - INFO - status of csicephfs-demo-pod at column STATUS was ContainerCreating, but we were waiting for Running
15:04:10 - MainThread - ocs_ci.utility.utils - INFO - Going to sleep for 3 seconds before next iteration
15:04:13 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod csicephfs-demo-pod -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 -o yaml
15:04:14 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod csicephfs-demo-pod -n namespace-test-b4970d6205b54e26a90a202cf2ab4895
15:04:15 - MainThread - tests.helpers - INFO - Pod csicephfs-demo-pod reached state Running
15:04:15 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod csicephfs-demo-pod -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 -o yaml
15:04:15 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod csicephfs-demo-pod -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 -o yaml
15:04:16 - MainThread - test_pod_from_registry - INFO - pod uses image-registry.openshift-image-registry.svc:5000/namespace-test-b4970d6205b54e26a90a202cf2ab4895/fio@sha256:39b639c8118135aae6923f6cd55df7c7030a901a4621cb6e712dda57bd9dec01
PASSED                                                                                                                                                                                                      [100%]
------------------------------------------------------------------------------------------------ live log teardown ------------------------------------------------------------------------------------------------
15:04:16 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig delete Pod csicephfs-demo-pod
15:04:49 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod csicephfs-demo-pod -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 -o yaml
15:04:50 - MainThread - ocs_ci.ocs.ocp - WARNING - Failed to get resource: csicephfs-demo-pod of kind: Pod, selector: None, Error: Error during execution of command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod csicephfs-demo-pod -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 -o yaml.
Error is Error from server (NotFound): pods "csicephfs-demo-pod" not found

15:04:50 - MainThread - ocs_ci.ocs.ocp - WARNING - Number of attempts to get resource reached!
15:04:50 - MainThread - ocs_ci.ocs.ocp - INFO - Pod csicephfs-demo-pod got deleted successfully
15:04:50 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get PersistentVolumeClaim pvc-test-83e89e48b28444198e0597f199725708 -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 -o yaml
15:04:50 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get PersistentVolume pvc-484ffddb-904a-406c-afb9-bb7c9ed42ccc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 -o yaml
15:04:51 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig delete PersistentVolumeClaim pvc-test-83e89e48b28444198e0597f199725708
15:04:52 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get PersistentVolumeClaim pvc-test-83e89e48b28444198e0597f199725708 -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 -o yaml
15:04:53 - MainThread - ocs_ci.ocs.ocp - WARNING - Failed to get resource: pvc-test-83e89e48b28444198e0597f199725708 of kind: PersistentVolumeClaim, selector: None, Error: Error during execution of command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get PersistentVolumeClaim pvc-test-83e89e48b28444198e0597f199725708 -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 -o yaml.
Error is Error from server (NotFound): persistentvolumeclaims "pvc-test-83e89e48b28444198e0597f199725708" not found

15:04:53 - MainThread - ocs_ci.ocs.ocp - WARNING - Number of attempts to get resource reached!
15:04:53 - MainThread - ocs_ci.ocs.ocp - INFO - PersistentVolumeClaim pvc-test-83e89e48b28444198e0597f199725708 got deleted successfully
15:04:53 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get PersistentVolume pvc-484ffddb-904a-406c-afb9-bb7c9ed42ccc -o yaml
15:04:54 - MainThread - ocs_ci.ocs.ocp - WARNING - Failed to get resource: pvc-484ffddb-904a-406c-afb9-bb7c9ed42ccc of kind: PersistentVolume, selector: None, Error: Error during execution of command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get PersistentVolume pvc-484ffddb-904a-406c-afb9-bb7c9ed42ccc -o yaml.
Error is Error from server (NotFound): persistentvolumes "pvc-484ffddb-904a-406c-afb9-bb7c9ed42ccc" not found

15:04:54 - MainThread - ocs_ci.ocs.ocp - WARNING - Number of attempts to get resource reached!
15:04:54 - MainThread - ocs_ci.ocs.ocp - INFO - PersistentVolume pvc-484ffddb-904a-406c-afb9-bb7c9ed42ccc got deleted successfully
15:04:54 - MainThread - ocs_ci.ocs.ocp - INFO - Switching to project openshift-storage
15:04:54 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc project openshift-storage
15:04:55 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig delete Project namespace-test-b4970d6205b54e26a90a202cf2ab4895
15:04:56 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Project namespace-test-b4970d6205b54e26a90a202cf2ab4895 -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 -o yaml
15:04:59 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Project namespace-test-b4970d6205b54e26a90a202cf2ab4895 -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 -o yaml
15:05:03 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Project namespace-test-b4970d6205b54e26a90a202cf2ab4895 -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 -o yaml
15:05:04 - MainThread - ocs_ci.ocs.ocp - WARNING - Failed to get resource: namespace-test-b4970d6205b54e26a90a202cf2ab4895 of kind: Project, selector: None, Error: Error during execution of command: oc -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Project namespace-test-b4970d6205b54e26a90a202cf2ab4895 -n namespace-test-b4970d6205b54e26a90a202cf2ab4895 -o yaml.
Error is Error from server (NotFound): namespaces "namespace-test-b4970d6205b54e26a90a202cf2ab4895" not found

15:05:04 - MainThread - ocs_ci.ocs.ocp - WARNING - Number of attempts to get resource reached!
15:05:04 - MainThread - ocs_ci.ocs.ocp - INFO - Project namespace-test-b4970d6205b54e26a90a202cf2ab4895 got deleted successfully
15:05:04 - Dummy-1 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Pod  -A -o yaml
15:05:04 - Dummy-8 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get StorageClass  -A -o yaml
15:05:04 - Dummy-9 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get CephFileSystem  -A -o yaml
15:05:04 - Dummy-4 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get CephBlockPool  -A -o yaml
15:05:04 - Dummy-5 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get PersistentVolume  -A -o yaml
15:05:04 - Dummy-7 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get PersistentVolumeClaim  -A -o yaml
15:05:04 - Dummy-6 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/kramdoss/Downloads/kramdoss-cluster/auth/kubeconfig get Namespace  -A -o yaml


==========================================================================================
```

ebenahar commented on tests/e2e/workloads/test_pod_from_registry.py:
```diff
@@ -0,0 +1,40 @@
+        image_id = image_obj.get().get('status').get('tags')[0].get('items')[0].get('dockerImageReference')
+        pod_dict = templating.load_yaml(constants.CSI_CEPHFS_POD_YAML)
+        pod_dict['spec']['containers'][0]['image'] = image_obj.resource_name
+        pod_dict['spec']['volumes'][0]['persistentVolumeClaim']['claimName'] = pvc_obj._name
```
please use pvc_obj.name and not _name

keesturam replied:
Made this change

ebenahar commented on tests/e2e/workloads/test_pod_from_registry.py:
```diff
@@ -0,0 +1,40 @@
+        pod_dict['spec']['volumes'][0]['persistentVolumeClaim']['claimName'] = pvc_obj._name
+        pod_obj = pod_factory(pvc=pvc_obj, custom_data=pod_dict)
+        pod_image = pod_obj.get().get('status').get('containerStatuses')[0].get('imageID')
+        if image_id == pod_image:
```
consider using assert, lines 37-40 could be replaced with:
```python
assert image_id == pod_image, f'pod uses different image {pod_image}'
logger.info(f'pod uses {image_id}')
```

keesturam replied:
I made this change as per your suggestion.

ebenahar commented on tests/helpers.py:
```diff
@@ -873,6 +877,101 @@ def create_pods(pvc_objs, pod_factory, interface, pods_for_rwx=1, status=""):
 
 
+def create_build_from_docker_image(
+        image_name,
```
delete extra indentation

keesturam replied:
Done

ebenahar commented on tests/helpers.py:
```diff
@@ -873,6 +877,101 @@ def create_pods(pvc_objs, pod_factory, interface, pods_for_rwx=1, status=""):
+        install_package (str): package to install over the base image
+
+    Returns:
+        (OCP): Returns the OCP object for the image
```
Returns:
    OCP: ....

keesturam replied:
changed

RazTamir commented on tests/e2e/workloads/test_pod_from_registry.py:
```diff
@@ -0,0 +1,37 @@
+        pod_dict['spec']['containers'][0]['image'] = image_obj.resource_name
+        pod_dict['spec']['volumes'][0]['persistentVolumeClaim']['claimName'] = pvc_obj.name
+        pod_obj = pod_factory(pvc=pvc_obj, custom_data=pod_dict)
+        pod_image = pod_obj.get().get('status').get('containerStatuses')[0].get('imageID')
```
I'm missing a call to check the pod is running with this new image

keesturam replied:
As discussed offline, this is validated by the pod factory.
